pkg/cmd/cli/cmd/set/buildsecret.go

```diff
@@ -0,0 +1,289 @@
+package set
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/errors"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	cmdutil "github.com/openshift/origin/pkg/cmd/util"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+const (
+	buildSecretLong = `
+Set or remove a build secret on a build config
+
+A build config can reference a secret to push or pull images from private registries or
+to access private source repositories.
+
+Specify the type of secret being set by using the --push, --pull, or --source flags.
+A secret reference can be removed by using --remove flag.
+
+A label selector may be specified with the --selector flag to select the build configs
+on which to set or remove secrets. Alternatively, all build configs in the namespace can
+be selected with the --all flag.
+`
+
+	buildSecretExample = `  # Clear push secret on a build config
+  %[1]s build-secret --push --remove bc/mybuild
+
+  # Set the pull secret on a build config
+  %[1]s build-secret --pull bc/mybuild mysecret
+
+  # Set the push and pull secret on a build config
+  %[1]s build-secret --push --pull bc/mybuild mysecret
+
+  # Set the source secret on a set of build configs matching a selector
+  %[1]s build-secret --source -l app=myapp gitsecret
+`
+)
+
+type BuildSecretOptions struct {
+	Out io.Writer
+	Err io.Writer
+
+	Builder *resource.Builder
+	Infos   []*resource.Info
+
+	Encoder       runtime.Encoder
+	OutputVersion unversioned.GroupVersion
+
+	Filenames []string
+	Selector  string
+	All       bool
+
+	ShortOutput bool
+	Mapper      meta.RESTMapper
+
+	PrintObject func(runtime.Object) error
+
+	Secret string
+	Push   bool
+	Pull   bool
+	Source bool
+	Remove bool
+}
+
+// NewCmdBuildSecret implements the set build-secret command
+func NewCmdBuildSecret(fullName string, f *clientcmd.Factory, out, errOut io.Writer) *cobra.Command {
+	options := &BuildSecretOptions{
+		Out: out,
+		Err: errOut,
+	}
+	cmd := &cobra.Command{
+		Use:     "build-secret BUILDCONFIG SECRETNAME",
```

Comment:
Is this structured in a way that allows me to later add support for `oc build-secret build/foo` without breaking backward compatibility?